source/FilesServer/app/service.py

```diff
@@ -22,10 +21,31 @@ def is_file_exists(
     return True
 
 
-async def s3_get_presigned_url(
-    file_name: str | None, s3_client: boto3.client = Depends(get_s3_client)
-) -> str | None:
+def s3_get_head_object_HTTP_headers(file_name: str) -> dict | None:
     settings = get_settings()
+    s3_client = get_s3_client()
+    if not is_file_exists(file_name):
+        return None
+    return s3_client.head_object(Bucket=settings.aws_bucket_name, Key=file_name)[
+        "ResponseMetadata"
+    ]["HTTPHeaders"]
+
+
+async def s3_get_metadata(file_name: str) -> dict | None:
```

Comment:
this comment is not redundant but FYI, if you annotate a return value (or param for that matter) with list or dict or types like that, please add the specific type of list or dict, etc. 
e.g. list[int], dict[int, str] (integer keys and string values)

Reply:
modified it to return FileMetadata